src/bridges/usdcBridge.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  BigNumberish,
+  BytesLike,
+  ethers,
+  AbiCoder,
+  ContractTransactionResponse,
+  Overrides,
+} from 'ethers';
+import {
+  Address,
+  PaymasterParams,
+  PriorityOpResponse,
+  TransactionResponse,
+} from '../types';
+import {IL1Nullifier__factory} from '../typechain';
+import {
+  REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT,
+  checkBaseCost,
+  scaleGasLimit,
+} from '../utils';
+import {Wallet} from '../wallet';
+import {insertGasPrice} from '../adapters';
+
+interface IDepositTransaction {
+  token: Address;
+  amount: BigNumberish;
+  to?: Address;
+  operatorTip?: BigNumberish;
+  bridgeAddress: Address;
+  approveERC20?: boolean;
+  l2GasLimit?: BigNumberish;
+  gasPerPubdataByte?: BigNumberish;
+  refundRecipient?: Address;
+  overrides?: Overrides;
+  approveOverrides?: Overrides;
+}
+
+interface IWithdrawTransaction {
+  token: Address;
+  amount: BigNumberish;
+  to?: Address;
+  bridgeAddress: Address;
+  paymasterParams?: PaymasterParams;
+  overrides?: Overrides;
+  approveERC20?: boolean;
+  approveOverrides?: Overrides;
+}
+
+export class USDCBridge {
+  constructor(protected readonly wallet: Wallet) {}
+
+  private getSecondBridgeDepositCalldata(
+    transaction: IDepositTransaction
+  ): string {
+    return AbiCoder.defaultAbiCoder().encode(
+      ['address', 'uint256', 'address'],
+      [transaction.token, transaction.amount, transaction.to]
+    );
+  }
+  /**
+   * @inheritDoc
+   *
+   * @example Deposit USDC.
+   *
+   * import { Wallet, Provider, types, USDCBridge } from "zksync-ethers";
+   * import { ethers } from "ethers";
+   *
+   * const PRIVATE_KEY = "<WALLET_PRIVATE_KEY>";
+   * const USDC_TOKEN_L1_ADDRESS = "<USDC_TOKEN_ADDRESS>";
+   * const USDC_BRIDGE_L1_ADDRESS = "<USDC_BRIDGE_L1_ADDRESS>";
+   * const AMOUNT = "5";
+   *
+   * const provider = Provider.getDefaultProvider(types.Network.Sepolia);
+   * const ethProvider = ethers.getDefaultProvider("sepolia");
+   * const wallet = new Wallet(PRIVATE_KEY, provider, ethProvider);
+   * const usdcBridge = new USDCBridge(wallet);
+   *
+   * const depositTx = await usdcBridge.deposit({
+   *   token: USDC_TOKEN_L1_ADDRESS,
```

Comment:
Assuming that bridge interface will be abstract, it does make sense to pass token as an argument, but if it's provided, maybe it makes sense to make sure that token is correct (e.g. at least check that token symbol is `USDC`)?

Reply:
I've made it a bit more generic and added `validateDepositParams` and `validateWithdrawParams` to the abstract `Bridge` class. `USDCBridge` overrides them and checks for the correct token addresses. Other bridges can implement them as needed.